### DISCUSSION TITLE
#79 Intended Solution?

### DISCUSSION THREAD

**Discussion by hahagu** [cssbattle/forum#61] [Q&A]
So, I'm using CSS Battle as a tool to practice my CSS skills rather than to get an optimal score.
However, I've been struggling on #79, and have no idea on how to get 100%.
Mostly, it came down to `overflow: hidden` and borders and box shadows, but I couldn't get it to work, even though in my brain it seems perfectly logical.

<details>
  <summary>Spoiler! Code Here!</summary>

```html
<div class="wrapper">
    <div class="circle mask">
        <div class="circle circle-primary"></div>
        <div class="circle inset"></div>
    </div>
    <div class="circle small smc-1"></div>
    <div class="circle small smc-2"></div>
    <div class="circle small smc-3"></div>
    <div class="circle small smc-4"></div>
    <div class="circle small smc-5"></div>
</div>

<style>
    body {
        background-color: #6592CF;
    }

    .wrapper {
        display: flex;
        align-items: center;
        justify-content: center;
        
        width: 100%;
        height: 100%;
    }

    .mask {
        display: flex;
        align-items: center;
        justify-content: center;
        
        height: 170px;
        width: 170px;
        margin-top: 20px;
        
      	overflow: hidden;
        
        border-radius: 50%;
    }

    .circle-primary {
        position: absolute;

        height: 150px;
        width: 150px;

        background-color: #6592CF;

        border-radius: 50%;

        border-style: solid;
        border-width: 10px;
        border-color: #243D83;
    }

    .inset {
        height: 100px;
        width: 100px;
      
      	background-color: #6592CF;
        
        border-radius: 50%;
      
        border-style: solid;
        border-width: 10px;
        border-color: #243D83;
      
      	margin-top: 200px;
      
      	z-index: 1;
    }

    .small {
        position: absolute;
        
        width: 10px;
        height: 10px;

        background-color: #243D83;

        border-radius: 50%;

        border-style: solid;
        border-width: 10px;
        border-color: #6592CF;
        
        box-shadow: 0 0 0 10px #243D83;
    }

    .smc-1 {
        margin: -140 0 0 0;
    }

    .smc-2 {
        margin: -60 140 0 0;
    }

    .smc-3 {
        margin: -60 -140 0 0;
    }

    .smc-4 {
        margin: 0 140 -80 0;
    }

    .smc-5 {
        margin: 0 -140 -80 0;
    }
</style>
```

</details>

This code leaves me with a very thin line separating the inset's bottom from the background, which doesn't really make sense to me.. Can't get it removed, and I tried many many things..

I also would, if possible, know some techniques to optimize the code if possible. right now, the code is messy, but considering I can't loop over stuff, I'd say I did a decent job in my own terms, but please provide feedback so I can improve.

There has to be a better way to do this, right..?

**Comment by SelenIT** [comment]
This target turns out really tough to get 100% match (only 46 players have managed to do this so far). I guess that anti-aliasing artifacts produce virtually invisible mismatching pixels near the intersections of large and small circles. Tweaking the `border-radius` of small circles seems to help sometimes. I managed to get your code to 100% with two changes:

- replacing `border-radius: 50%` for `.small` with `border-radius: 48%`;
- overlaying that unwanted thin line with the second shadow of the top small circle, by replacing `box-shadow: 0 0 0 10px #243D83` for `.small` with ` box-shadow: 0 0 0 10px #243D83, 0 180px 0 35px #6592CF;`,

Maybe this thin line can be removed also by tweaking the size and/or `border-radius` of `.mask` by fractions of pixels, but I had no success with this approach.

Regarding optimization techniques, many of them are listed on the [Tips & Tricks](https://cssbattle.dev/tips/) page and in the [Tips'n'tricks thread](https://spectrum.chat/css-battle/general/tips-n-tricks~6034b7fa-d08b-463a-9720-decb5e2c0f5b) in the old Spectrum community.

**Reply by hahagu** [reply]
I've tried, and it does work.
However, I have trouble understanding the logic, as the line was 'leaking' from `.circle-primary`, and `.small` contains the properties for the small circles around and above (in terms of z axis) the primary circle.
How does this come to affect the thin border?

Also, after your change, the `border-radius` to be specific, the target does not visually match. How does this come to be, and why is it 100%?
Thanks. 

Edit: I think it also should be added, that I've tried the approach of tweaking the `border-radius` of `.mask` by fractions as well without success.
